compose/config/config.py

```diff
@@ -495,18 +488,10 @@ def resolve_volume_path(volume, working_dir, service_name):
     container_path = os.path.expanduser(container_path)
 
     if host_path is not None:
-        if not any(host_path.startswith(c) for c in PATH_START_CHARS):
-            log.warn(
-                'Warning: the mapping "{0}:{1}" in the volumes config for '
-                'service "{2}" is ambiguous. In a future version of Docker, '
-                'it will designate a "named" volume '
-                '(see https://github.com/docker/docker/pull/14242). '
-                'To prevent unexpected behaviour, change it to "./{0}:{1}"'
-                .format(host_path, container_path, service_name)
-            )
-
+        if host_path.startswith('.'):
```

Comment:
No, it'll still get expanded - `~` is handled by the `os.path.expanduser` line below.